persistence/mongodb-record/src/main/scala/net/liftweb/mongodb/record/field/BsonRecordField.scala

```diff
@@ -27,59 +27,70 @@ import net.liftweb.record._
 import com.mongodb._
 import org.bson.Document
 
+import scala.reflect.Manifest
 import scala.xml._
 
 /** Field that contains an entire record represented as an inline object value. Inspired by JSONSubRecordField */
-class BsonRecordField[OwnerType <: BsonRecord[OwnerType], SubRecordType <: BsonRecord[SubRecordType]]
-                        (rec: OwnerType, valueMeta: BsonMetaRecord[SubRecordType])(implicit subRecordType: Manifest[SubRecordType])
-  extends Field[SubRecordType, OwnerType]
-  with MandatoryTypedField[SubRecordType]
-{
-  def this(rec: OwnerType, valueMeta: BsonMetaRecord[SubRecordType], value: SubRecordType)
-          (implicit subRecordType: Manifest[SubRecordType]) = {
-      this(rec, value.meta)
-      set(value)
-  }
-
-  def this(rec: OwnerType, valueMeta: BsonMetaRecord[SubRecordType], value: Box[SubRecordType])
-          (implicit subRecordType: Manifest[SubRecordType]) = {
-      this(rec, valueMeta)
-      setBox(value)
+abstract class BsonRecordTypedField[OwnerType <: BsonRecord[OwnerType], SubRecordType <: BsonRecord[SubRecordType]]
+(override val owner: OwnerType, valueMeta: BsonMetaRecord[SubRecordType])(implicit subRecordType: Manifest[SubRecordType])
+  extends Field[SubRecordType, OwnerType] {
+
+  def this(owner: OwnerType, valueMeta: BsonMetaRecord[SubRecordType], value: Box[SubRecordType])
+    (implicit subRecordType: Manifest[SubRecordType]) = {
+    this(owner, valueMeta)
+    setBox(value)
   }
 
-  def owner = rec
-  def asJs = asJValue match {
+  override def asJs = asJValue match {
```

Comment:
I'm curious as to why `override` was added to these functions? The functions are abstract in the parent class, so `override` is not needed.

Reply:
Oh, I thought it makes it clear that we are implementing/overriding something that's coming from the super-class (or trait). Also, It has happened to me once or twice that when refactoring a method, it makes it quick to spot the error if you accidentally change the type of what you are returning from the method (although, in this case, if the class itself is not abstract, the compiler will complain anyway). Just being a general smart-ass. Can roll it back if necessary.

Reply:
I guess I prefer to not use override unless it's actually overriding something.

Reply:
I would prefer that, but it's not that big of a deal and I'm not sure if others have any opinions on it.

Reply:
I would also prefer that we revert that piece of the change :)

Reply:
Okay. One commit coming right along. :smiley: